app/views/_partials/acmsh/acmsh.ejs

```diff
@@ -0,0 +1,140 @@
+<script type="text/javascript">
+class ACMsh {
+    constructor() {
+        this.input = "";
+        this.display = "";
+        this.cursor_alt = false;
+        this.history = [''];
+        this.history_pos = 0;
+        this.cursor_pos = 0
+
+        window.addEventListener("keydown", function(e) {
+            // space and arrow keys
+            if([' ', 'Backspace', 'ArrowUp', 'ArrowLeft', 'ArrowRight', 'ArrowDown'].indexOf(e.key) > -1) {
+                e.preventDefault();
+            }
+        }, false);
+
+        document.addEventListener('keydown', (event) => {
+            switch (event.key) {
+                case "ArrowUp":
+                    if (this.history_pos < this.history.length) {
+                        this.history_pos += 1;
+                        this.input = this.history[this.history.length - this.history_pos];
+                    }
+                    break;
+
+                case "ArrowDown":
+                    if (this.history_pos > 0) {
+                        this.history_pos -= 1;
+                        var next = this.history.length - this.history_pos;
+                        this.history.push("")
+                        this.input = this.history[next];
+                        this.history.pop();
+                    }
+                    break;
+                
+                case "ArrowLeft":
+                    if (this.cursor_pos > 0) {
+                        this.cursor_pos -= 1
```

Comment:
I'm gonna make an issue about this, but we should probably have linting for frontend.

Can you add semicolons to the end of the lines though in the meantime?